go-controller/pkg/libovsdbops/switch.go

```diff
@@ -222,7 +222,28 @@ func RemoveACLsFromLogicalSwitchesWithPredicateOps(nbClient libovsdbclient.Clien
 // RemoveACLsFromLogicalSwitchesWithPredicate looks up logical switches from the cache
 // based on a given predicate and removes from them the provided ACLs
 func RemoveACLsFromLogicalSwitchesWithPredicate(nbClient libovsdbclient.Client, p switchPredicate, acls ...*nbdb.ACL) error {
-	ops, err := RemoveACLsFromLogicalSwitchesWithPredicateOps(nbClient, nil, p, acls...)
+	// optimize the predicate to exclude switches that don't reference deleting acls.
+	// it is safe to do, since this function transacts the ops.
+	swWithACLsPred := func(sw *nbdb.LogicalSwitch) bool {
+		if !p(sw) {
+			return false
+		}
+		// in most cases switch is expected to have acls that are being deleted, so this should return fast.
+		// we iterate acls first, in case switch has some other acls.
+		// one exception is deleting a lot of acls as a cleanup, that will result in an empty transaction for most cases.
+		// that would mean len(acls)*len(sw.ACLs) iterations, but we expect sw.ACLs to be close to empty in that case.
+		// if that assumption changes in the future, the loop may be optimized by creating a set/map of acls UUIDs for
+		// faster comparison.
+		for _, deleteACL := range acls {
+			for _, swACL := range sw.ACLs {
```

Comment:
Consider having this predicate outside, in egressfirewall.go. 

That predicate implementation could be pretty clean, like
```
aclsToDelete := sets.NewString(aclUUIDs)
p := func(item *nbdb.LogicalSwitch) bool {
    return aclsToDelete.HasAny(item.ACLs)
}
```